app/models/peer_review.rb

```diff
@@ -0,0 +1,16 @@
+class PeerReview < ActiveRecord::Base
+  belongs_to :result
+  belongs_to :reviewer, class_name: 'Grouping'
+
+  validates_associated :reviewer
+  validates_associated :result
+  validates_presence_of :reviewer
+  validates_presence_of :result
+  validates_numericality_of :reviewer_id, only_integer: true, greater_than: 0
+  validates_numericality_of :result_id, only_integer: true, greater_than: 0
+
+  def reviewee
+    # TODO - Research optimizing or see if rails can do better
+    Grouping.joins({ :submissions => { :results => :peer_reviews }}).where('peer_reviews.id = ?', self.id).first
```

Comment:
I'm not too happy with this, however it works.
I guess the decisions is, are you okay with it? Is it worth merging it since it works, and then trying to find a better way if such a way exists? The internet was conflicted on what was possible and I had more difficulty than I'd like to admit getting other hacky solutions to work.